stable/restore/files/nuorestore

```diff
@@ -125,12 +144,60 @@ done
 restore_type=$(echo $restore_type | tr '[:upper:]' '[:lower:]')
 auto=$(echo $auto | tr '[:upper:]' '[:lower:]')
 
+if ! isUrl "$restore_source"; then
+   # Resolve special restore sources - ":latest", "<backup_group>:latest" or
+   # "<backup_group>:<index>"; fail the database restore early if we are not
+   # able to do that
+   if echo "$restore_source" | grep -Eq ":latest$"; then
+      backup_group="${restore_source%:latest}"
+      if [ -z "$backup_group" ]; then
+         backup_group=$(nuocmd get value --key "$NUODB_BACKUP_KEY/$db_name/latest")
+      fi
+      backup_index=$(nuocmd get value --key "$NUODB_BACKUP_KEY/$db_name/$backup_group/latest")
+      if [ -z "$backup_index" ]; then
+         die 1 "Unable to find the latest backup for backup group '${backup_group}'"
```

Comment:
Single-quoting `'${backup_group}'` specifies it as a literal. Is that what you want?

Reply:
Same comment on lines 164 and 173.

Reply:
Actually, the shell with expanding the variable as the whole string is in double quotes. This is a quick test to demonstrate it.

```shell
bash-3.2$ backup_group=cluster0
bash-3.2$ die 1 "Unable to find the latest backup for backup group '${backup_group}'"
Unable to find the latest backup for backup group 'cluster0'
```

And here are some tests with the actual script:

```shell
$ k logs restore-demo-n2qff -f
Unable to connect to http://nuodb.nuodb.svc:8888: HTTPConnectionPool(host='nuodb.nuodb.svc', port=8888): Max retries exceeded with url: /api/1/kvstore/%2Fnuodb%2Fnuobackup%2Fdemo%2Ffoo%2F2 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fbd50f56a10>: Failed to establish a new connection: [Errno -2] Name or service not known',))
2022-01-12T11:24:07.459+0000 Unable to find a valid backup set for restore source 'foo:2'
```

```shell
$ k logs restore-demo-6nrqq -f
2022-01-12T11:29:10.704+0000 Unable to find the latest backup for backup group 'foo'
```

Reply:
I see. It seems like I never stop learning new and surprising things about shell scripting.